src/app/SIPPacketMangler.cs

```diff
@@ -37,13 +37,24 @@ public static string MangleSDP(string sdpBody, string publicIPAddress, out bool
             {
                 if (sdpBody != null && publicIPAddress != null)
                 {
-                    string sdpAddress = SDP.GetSDPRTPEndPoint(sdpBody).Address.ToString();
+                    IPAddress addr = SDP.GetSDPRTPEndPoint(sdpBody).Address;
+                    //rj2: need to consider publicAddress and IPv6 for mangling
+                    IPAddress pubaddr = IPAddress.Parse(publicIPAddress);
+                    string sdpAddress = addr.ToString();
 
+                    if (IPSocket.IsPrivateAddress(sdpAddress) && publicIPAddress != sdpAddress && addr.AddressFamily == System.Net.Sockets.AddressFamily.InterNetworkV6)
+                    {
+                        string mangledSDP = Regex.Replace(sdpBody, @"c=IN IP6 (?<ipaddress>([:a-fA-F0-9]+))", "c=IN " + (pubaddr.AddressFamily == System.Net.Sockets.AddressFamily.InterNetworkV6 ? "IP6 " : "IP4 ") + publicIPAddress, RegexOptions.Singleline | RegexOptions.ExplicitCapture);
+                        wasMangled = true;
+
+                        return mangledSDP;
+                    }
+                    else
                     // Only mangle if there is something to change. For example the server could be on the same private subnet in which case it can't help.
                     if (IPSocket.IsPrivateAddress(sdpAddress) && publicIPAddress != sdpAddress)
                     {
                         //logger.LogDebug("MangleSDP replacing private " + sdpAddress + " with " + publicIPAddress + ".");
-                        string mangledSDP = Regex.Replace(sdpBody, @"c=IN IP4 (?<ipaddress>(\d+\.){3}\d+)", "c=IN IP4 " + publicIPAddress, RegexOptions.Singleline);
+                        string mangledSDP = Regex.Replace(sdpBody, @"c=IN IP4 (?<ipaddress>(\d+\.){3}\d+)", "c=IN " + (pubaddr.AddressFamily == System.Net.Sockets.AddressFamily.InterNetworkV6 ? "IP6 " : "IP4 ") + publicIPAddress, RegexOptions.Singleline);

```

Comment:
I don't think this will work as expected. If the original SDP offer has a private IPv4 address then replacing it with a public IPv6 address is unlikely to result in the RTP packets getting to the destination. IPv4 private IP addresses could be replaced with IPv4 public addresses and the same for IPv6.

Reply:
```suggestion
                    if (IPSocket.IsPrivateAddress(sdpAddress) && publicIPAddress != sdpAddress && addr.AddressFamily == pubaddr.AddressFamily && pubaddr.AddressFamily == System.Net.Sockets.AddressFamily.InterNetwork)
                    {
                        //logger.LogDebug("MangleSDP replacing private " + sdpAddress + " with " + publicIPAddress + ".");
                        string mangledSDP = Regex.Replace(sdpBody, @"c=IN IP4 (?<ipaddress>(\d+\.){3}\d+)", "c=IN IP4 " + publicIPAddress, RegexOptions.Singleline);

```
changing the (else-if)condition should solve this?

Reply:
The if/else will partly solve it. But the regex also has "c=IN IP4..." which is never going to match on an SDP offer with an IPv6 address.